functions/utils/getRequest.ts

```diff
@@ -0,0 +1,31 @@
+import { MetaTagInjector } from '../components/metaTagInjector'
+import Cache from './cache'
+
+export default async function getRequest(
+  res: Promise<Response>,
+  url: string,
+  getData: () => Promise<
+    | {
+        title: string
+        image: string
+        url: string
+      }
+    | undefined
+  >
+) {
+  try {
+    const cacheResponse = await Cache.match(url)
```

Comment:
```suggestion
    const cachedData = await Cache.match(url)
```

to match `data` and make it obvious that they are the same type.